tests/system_suite_test.go

```diff
@@ -0,0 +1,95 @@
+//go:build !unit
+// +build !unit
+
+package tests
+
+import (
+	"flag"
+	"path"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	apiext "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	k8sRuntime "k8s.io/apimachinery/pkg/runtime"
+	ctlr "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/nginxinc/nginx-gateway-fabric/tests/framework"
+)
+
+func TestNGF(t *testing.T) {
+	flag.Parse()
+	if *gatewayAPIVersion == "" {
+		panic("Gateway API version must be set")
+	}
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "NGF System Tests")
+}
+
+var k8sClient client.Client
+
+var _ = BeforeSuite(func() {
```

Comment:
I wonder if we need to install NGF and Gateway API in BeforeSuite at all.


What if we do it in every test? I'm thinking about easy of customization and level of control.

So something like this:
```go
ar _ = Describe("tests", func() {
	var (
		gwAPI ...
		ngf ...
	)
	BeforeAll(func() {
		gwAPI = InstallGWAPI(
			WithVersion("v0.3.0"), // customizing version
			)
		ngf = InstallNGF(
			WithReplicaCount(3), // customizing replica count
			)
	})

	Describe("my tests", func() {
		// tests
	})

	AfterAll(func() {
		UninstallGWAPI(gwAPI)
		UninstallNGF(ngf)
	})
})
```

At the same time, if there multiple test files in a suite that share the same NGF and NGF, they can optionally do the same thing in BeforeSuite() and AfterSuite().

Reply:
Lot of overhead to do it in every test, I suspect that we either split up into separate suites depending on the configuration, or utilize Contexts and deploy NGF at that level for common tests.